packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue

```diff
@@ -1,81 +1,107 @@
 <template>
-	<div
-		v-if="matrix"
-		class="p-datatable p-component p-datatable-scrollable p-datatable-responsive-scroll p-datatable-gridlines p-datatable-grouped-header stratified-value-matrix"
-	>
-		<div class="p-datatable-wrapper">
-			<table class="p-datatable-table p-datatable-scrollable-table editable-cells-table">
-				<thead v-if="nodeType === NodeType.Transition" class="p-datatable-thead">
-					<tr>
-						<th class="choose-criteria"></th>
-						<th class="choose-criteria"></th>
-						<th v-for="i in matrix[0].length + 1" :key="i" class="choose-criteria">
-							<Dropdown
-								v-if="i === 1"
-								class="w-full"
-								placeholder="Select a variable"
-								v-model="chosenCol"
-								:options="colDimensions"
-							/>
-						</th>
-					</tr>
-					<tr>
-						<th class="choose-criteria"></th>
-						<th class="choose-criteria"></th>
-						<th v-for="(row, i) in matrix[0]" :key="i">
-							{{ row.colCriteria?.[chosenCol] }}
-						</th>
-					</tr>
-				</thead>
-				<tbody class="p-datatable-tbody">
-					<tr v-for="(row, i) in matrix" :key="i">
-						<td class="p-frozen-column">
-							<Dropdown
-								v-if="i === 0"
-								class="w-full"
-								placeholder="Select a variable"
-								v-model="chosenRow"
-								:options="rowDimensions"
-							/>
-						</td>
-						<td class="p-frozen-column">{{ row[0].rowCriteria?.[chosenRow] }}</td>
-						<td
-							v-for="(cell, j) in row"
-							:key="j"
-							@click="
-								valueToEdit = {
-									val: getMatrixValue(cell?.value?.id),
-									rowIdx: i,
-									colIdx: j
-								}
-							"
-						>
-							<template v-if="cell?.value?.id">
-								<InputText
-									v-if="valueToEdit.rowIdx === i && valueToEdit.colIdx === j"
-									class="cell-input"
-									v-model.lazy="valueToEdit.val"
-									v-focus
-									@focusout="valueToEdit = { val: '', rowIdx: -1, colIdx: -1 }"
-									@keyup.stop.enter="updateModelConfigValue(cell?.value?.id)"
+	<main>
+		<div
+			v-if="matrix"
+			class="p-datatable p-component p-datatable-scrollable p-datatable-responsive-scroll p-datatable-gridlines p-datatable-grouped-header stratified-value-matrix"
+		>
+			<div class="p-datatable-wrapper">
+				<table class="p-datatable-table p-datatable-scrollable-table editable-cells-table">
+					<thead v-if="nodeType === NodeType.Transition" class="p-datatable-thead">
+						<tr>
+							<th class="choose-criteria"></th>
+							<th class="choose-criteria"></th>
+							<th v-for="i in matrix[0].length + 1" :key="i" class="choose-criteria">
```

Comment:
I absolutely could be wrong, but, in this block of code you have `v-for"i in matrix[0].length+1`.  Wouldn't that just be scoped to elements inside that `<th` tag and whats inside of it (`<Dropdown...`)?  Down below in another `<tr><th>` tree you reference i which seems wrong or if its not, I'd love a quick primer on how that works